jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/command/AddDataCaseFileInstanceCommand.java

```diff
@@ -85,6 +87,16 @@ public Void execute(Context context) {
         
         CaseEventSupport caseEventSupport = getCaseEventSupport(context);
         caseEventSupport.fireBeforeCaseDataAdded(caseFile.getCaseId(), caseFile, caseFile.getDefinitionId(), parameters);
+        
+        org.jbpm.process.instance.ProcessInstance pi = (org.jbpm.process.instance.ProcessInstance) ksession.getProcessInstance(processInstanceId);
```

Comment:
logic spreaded between VariableScopeInstance and here.

Reply:
VariableScopeInstance is not really used in this scenario. Please check getVariable, which relies on CaseFileInnstan

Reply:
Code changed to reuse setVariable on VariableScopeInstance